src/Futhark/CLI/LSP.hs

```diff
@@ -0,0 +1,42 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Futhark.CLI.LSP (main) where
+
+import Control.Concurrent.MVar (newMVar)
+import Control.Monad.IO.Class (MonadIO (liftIO))
+import Futhark.LSP.Handlers (handlers)
+import Futhark.LSP.Utils (debug, emptyState)
+import Language.LSP.Server
+import Language.LSP.Types
+import System.Log.Logger (Priority (DEBUG))
+
+main :: String -> [String] -> IO ()
+main _prog _args = do
+  stateMVar <- newMVar emptyState
```

Comment:
We prefer `snake_case` for local non-function variables; see [the style guide](https://github.com/diku-dk/futhark/blob/master/STYLE.md#naming).